### PR TITLE
Add multimodel performance report

### PR DESCRIPTION
**Pull request by matanper**

### CODE REVIEW

File: deepchecks/base/check.py [hunk 1]
```diff
@@ -395,23 +397,34 @@ def run(self, model) -> CheckResult:
...
-            models: List[Any]
-            ) -> CheckResult:
-        """Preprocess the parameters and pass them to `_run`."""
+class ModelComparisonContext:
+    """Contain processed input for model comparison checks."""
+    def __init__(self,
+                 train_datasets: Union[Dataset, List[Dataset]],
+                 test_datasets: Union[Dataset, List[Dataset]],
+                 models: Union[List[Any], Mapping[str, Any]]
+                 ):
```
Comment (nirhutnik) [inline deepchecks/base/check.py:406]: Why is this a class and not a function? If this is mainly validations...

Comment (nirhutnik) [reply deepchecks/base/check.py:406]: Also, why not use a context for every train-test-model triplet and then just send as a list? Only because of the names thing?

**Comment by nirhutnik** [comment]
<p>Agree with Bressler. Also, maybe different colors? </p><p>And wondering whether on the x-axis, should the first degree be the classes and then the metrics (as is here)? I think it would make more sense to have the first degree be the metric and then the classes.</p>

---

 <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/300/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: deepchecks/base/check.py (hunk 1 above)
Comment (matanper) [reply deepchecks/base/check.py:406]: because I want extra info, and some of this extra info is shared. for example task_type is single, not every triplet has its own. 
Also, the name makes it a quadruplet, and for every little thing you will want to add in the future it will change to quintuple, and then to sextuple, septuple, octuple, up to milluple.

**Comment by matanper** [comment]
<p>Colors are part of the global theme, and I think also the height. gabbay played with it, we should ask him.</p><p>About x-axis, that was bressler strict orders.</p>

---

 <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/300/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by noamzbr** [comment]
<p>Nir's suggestion is not bad, let's try it?</p>

---

 <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/300/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by noamzbr** [comment]
<p>About the color and Hight - we should check </p>

---

 <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/300/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by matanper** [comment]
<p>Nir suggestions are the worst</p>

---

 <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/300/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: deepchecks/base/check.py (hunk 1 above)
Comment (nirhutnik) [reply deepchecks/base/check.py:406]: :+1:

**Comment by nirhutnik** [comment]
<p>If you think my suggestions are bad, you should see my code</p>

---

 <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/300/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: deepchecks/base/check.py [hunk 2]
```diff
@@ -431,10 +445,53 @@ def run(self,
...
+            train_datasets: Union[Dataset, List[Dataset]],
+            test_datasets: Union[Dataset, List[Dataset]],
+            models: Union[List[Any], Mapping[str, Any]]
+            ) -> CheckResult:
+        """Initialize context and pass to check logic."""
+        return self.run_logic(ModelComparisonContext(train_datasets, test_datasets, models))
 
     @abc.abstractmethod
-    def _run(self, datasets_models: List[Tuple[Dataset, Dataset, Any]]):
+    def run_logic(self, context: ModelComparisonContext):
```
Comment (noamzbr) [inline deepchecks/base/check.py:493]: Maybe "run_with_context" or "run_model_comparison_internal" ?

File: deepchecks/checks/performance/performance_report.py [hunk 3]
```diff
@@ -77,3 +80,59 @@ def condition(result, min_score):
             return ConditionResult(True)
 
         return self.add_condition(name, condition, min_score=min_score)
+
+
+class MultiModelPerformanceReport(ModelComparisonBaseCheck):
+    """Summarize given scores between models on test datasets.
```
Comment (noamzbr) [inline deepchecks/checks/performance/performance_report.py:86]:
```suggestion
    """Report score functions results for input models on test datasets.
```

File: deepchecks/utils/metrics.py [hunk 4]
```diff
@@ -89,6 +89,75 @@ class ModelType(enum.Enum):
 }
 
 
+class DeepcheckScorer:
+    """Encapsulate scorer function with extra methods.
```
Comment (noamzbr) [inline deepchecks/utils/metrics.py:93]: Explain here what a scorer is, refer to https://scikit-learn.org/stable/modules/model_evaluation.html

File: deepchecks/utils/metrics.py [hunk 5]
```diff
@@ -89,6 +89,75 @@ class ModelType(enum.Enum):
...
+                                                  f'returned only {len(result)} elements in the score array value')
+
+        else:
+            result = self._run_score(model, df.head(2), dataset)
+            if not isinstance(result, Number):
+                raise errors.DeepchecksValueError(f'Expected scorer {self.name} to return number '
+                                                  f'but got: {type(result).__name__}')
+
+    def is_negative_scorer(self):
+        return self.sklearn_scorer_name is not None and self.sklearn_scorer_name.startswith('neg_')
```
Comment (noamzbr) [inline deepchecks/utils/metrics.py:158]: Nice

**Review by noamzbr — COMMENTED** [review]
Remember to add an nblink file in docs for the new notebook

File: deepchecks/base/check.py (hunk 2 above)
Comment (matanper) [reply deepchecks/base/check.py:493]: that's ugly name, and this is the method that will be implemented in each check, so it can't be too ugly

File: deepchecks/checks/performance/performance_report.py (hunk 3 above)
Comment (matanper) [reply deepchecks/checks/performance/performance_report.py:86]: What's good about the word report? it weird-sounding as a verb

File: deepchecks/utils/metrics.py (hunk 4 above)
Comment (matanper) [reply deepchecks/utils/metrics.py:93]: Why? it's internal class

**Review by noamzbr — APPROVED** [review]

File: deepchecks/utils/metrics.py (hunk 4 above)
Comment (noamzbr) [reply deepchecks/utils/metrics.py:93]: Still people may wanna use it, but can solve in future PR
